react-ui/src/pages/submission/Submission.jsx

```diff
@@ -135,277 +38,46 @@ const useStyles = makeStyles((theme) => ({
     justifyContent: 'space-between',
     width: '100%',
   },
-  previewDescription: {
-    '& textarea': {
-      visibility: 'hidden',
-    },
-  },
-  previewDescriptionButton: {
-    alignSelf: 'flex-start',
-  },
-  submitAnotherButton: {
-    marginLeft: theme.spacing(2),
-  },
   title: {
     marginBottom: theme.spacing(2),
   },
 }));
 
-const fileReader = new FileReader();
-
 function Submission() {
+  // ??? Pass in to submission form or not?
   const {
     data: {
-      categories,
-      firebaseToken,
       id: contestId,
       name: contestName,
       prompt,
       submissionEnd,
       submissions = [],
     },
     error,
-  } = useSwrData(API_PATH);
-  const { isMutating, trigger } = useSwrMutation(API_PATH, postData);
+    isLoading,
+  } = useSwrSubmission();
+  // useSWRMutation shares cache store with useSWR to avoid race conditions
 
-  const [formState, updateFormState, resetFormState] = useFormState([
-    'name',
-    'category',
-    'file',
-    'description',
-    'complianceOriginalDesign',
-    'complianceAuthorshipAnonymous',
-    'complianceEffort',
-    'complianceOriginalArt',
-    'complianceNsfwFree',
-    'complianceFlatFlag',
-  ]);
-  const [{ username }] = useAuthState();
   const { state } = useLocation();
-  const updateSnackbarState = useSnackbarState();
   const [selectedTab, setSelectedTab] = useState(state?.defaultTab ?? 0);
-  const [fileDimensions, setFileDimensions] = useState(null);
   const [previewDescription, setPreviewDescription] = useState(false);
```

Comment:
I left it as required since `previewDescription` comes from `useState`, so if that's missing we've got a deeper issue. I did fix TextField, JSDoc, and propType though.

Next project will definitely be Typescript

Reply:
`isRequired` means the value provided is non-null and not undefined. If `useState`'s default is `null` this will fail the prop check in SubmissionForm.

Agreed that Typescript should make things easier. It should theoretically be possible to make new files in TSX while still having JSX, seems like it's just adding a `tsconfig.json`: https://www.totaltypescript.com/add-typescript-to-existing-react-project

Reply:
I think you highlighted / are looking at the wrong line above, `previewDimensions` is a boolean while `fileDimensions` is never used as a prop.